Tests/SourceKitTests/ImplementationTests.swift

```diff
@@ -0,0 +1,67 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+@testable import SourceKit
+import LanguageServerProtocol
+import XCTest
+import SKTestSupport
+import ISDBTestSupport
+
+final class ImplementationTests: XCTestCase {
+  func testImplementation() throws {
+    let ws = try staticSourceKitTibsWorkspace(name: "Implementation")!
+    try ws.buildAndIndex()
+
+    try ws.openDocument(ws.testLoc("a.swift").url, language: .swift)
+    try ws.openDocument(ws.testLoc("b.swift").url, language: .swift)
+
+    func impls(at testLoc: TestLocation) throws -> [Location] {
+      let textDocument = TextDocumentIdentifier(testLoc.url)
+      let request = ImplementationRequest(textDocument: textDocument, position: Position(testLoc))
+      let implementations = try ws.sk.sendSync(request)
+      return implementations
+    }
+    func testLoc(_ name: String) -> TestLocation {
+      ws.testLoc(name)
+    }  
+    func loc(_ name: String) -> Location {
+      Location(ws.testLoc(name))
+    }
+    
+    try XCTAssertEqual(impls(at: testLoc("Protocol")), [loc("StructConformance")])
+    try XCTAssertEqual(impls(at: testLoc("ProtocolStaticVar")), [loc("StructStaticVar")])
+    try XCTAssertEqual(impls(at: testLoc("ProtocolStaticFunction")), [loc("StructStaticFunction")])
+    try XCTAssertEqual(impls(at: testLoc("ProtocolVariable")), [loc("StructVariable")])
+    try XCTAssertEqual(impls(at: testLoc("ProtocolFunction")), [loc("StructFunction")])
+    try XCTAssertEqual(impls(at: testLoc("Class")), [loc("SubclassConformance")])
+    try XCTAssertEqual(impls(at: testLoc("ClassClassVar")), [loc("SubclassClassVar")])
+    try XCTAssertEqual(impls(at: testLoc("ClassClassFunction")), [loc("SubclassClassFunction")])
+    try XCTAssertEqual(impls(at: testLoc("ClassVariable")), [loc("SubclassVariable")])
+    try XCTAssertEqual(impls(at: testLoc("ClassFunction")), [loc("SubclassFunction")])
+
+    try XCTAssertEqual(impls(at: testLoc("Sepulcidae")), [loc("ParapamphiliinaeConformance"), loc("XyelulinaeConformance"), loc("TrematothoracinaeConformance")])
```

Comment:
The order of these is not guaranteed in the index.  Maybe we should change `impls` to return a `Set` for testing purposes?

Reply:
changed